snapcraft/cli/lifecycle.py

```diff
@@ -117,6 +117,10 @@ def _execute(  # noqa: C901
         if pack_project:
             _pack(project.prime_dir, output=output)
     else:
+        # load project and validate schema before invoking a provider
+        if project.info.base is not None:
+            project_config = project_loader.load_config(project)
```

Comment:
this will load a lot of code on the host, the host being Ubuntu or any other linux variant, Windows and OSX, we should think about alternatives to this.

Reply:
Changed to validate plugin schemas by directly calling the plugin loader.